components/properties/src/provider.rs

```diff
@@ -225,3 +44,133 @@ pub struct ScriptWithExtensionsPropertyV1<'data> {
     #[cfg_attr(feature = "serde", serde(borrow))]
     pub data: ScriptWithExtensions<'data>,
 }
+
+macro_rules! expand {
+    (
+        ($(($bin_marker:ident, $bin_s:literal),)+),
+        ($(($enum_marker:ident, $enum_s:literal, $value_ty:ident),)+)
+    ) => {
+
+            $(
+                pub struct $bin_marker;
+
+                impl DataMarker for $bin_marker {
+                    type Yokeable = UnicodePropertyV1<'static>;
+                }
+                impl ResourceMarker for $bin_marker {
+                    const KEY: ResourceKey = resource_key!(concat!("props/", $bin_s, "@1"));
+                }
+            )+
+
+            $(
+                pub struct $enum_marker;
+
+                impl DataMarker for $enum_marker {
+                    type Yokeable = UnicodePropertyMapV1<'static, crate::$value_ty>;
+                }
+
+                impl ResourceMarker for $enum_marker {
+                    const KEY: ResourceKey = resource_key!(concat!("props/", $enum_s, "@1"));
+                }
+            )+
+
+            #[cfg(feature = "datagen")]
+            /// The set of all resource keys supported by [`icu_uniset`](crate).
+            pub const ALL_KEYS: [ResourceKey; 75] = [
+                $(
+                    $bin_marker::KEY,
+                )+
+                $(
+                    $enum_marker::KEY,
+                )+
+                ScriptWithExtensionsPropertyV1Marker::KEY,
+            ];
+    };
+}
+
+expand!(
+    (
+        // Binary properties
+        (AsciiHexDigitV1Marker, "AHex"),
+        (AlnumV1Marker, "alnum"),
+        (AlphabeticV1Marker, "Alpha"),
+        (BidiControlV1Marker, "Bidi_C"),
+        (BidiMirroredV1Marker, "Bidi_M"),
+        (BlankV1Marker, "blank"),
+        (CasedV1Marker, "Cased"),
+        (CaseIgnorableV1Marker, "CI"),
+        (FullCompositionExclusionV1Marker, "Comp_Ex"),
+        (ChangesWhenCasefoldedV1Marker, "CWCF"),
```

Comment:
Key names are now decoupled from the file names used in datagen, so we could
* make them lower snake case like all other keys
* use longer, more readable names

Wdyt?

Reply:
The names are the official Unicode property names. All properties have a long name and most have a short name. Since we are no longer size-limited in the key names, I have no strong opinion about whether we should use the long names or the short names here.